app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizHistoryActivity.java

```diff
@@ -0,0 +1,66 @@
+package org.liberty.android.fantastischmemo.ui;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import org.liberty.android.fantastischmemo.R;
+import org.liberty.android.fantastischmemo.entity.History;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class QuizHistoryActivity extends AppCompatActivity {
+
+    public static final String EXTRA_DBPATH = "dbpath";
+
+    private TextView averageTextView;
+    private TextView attemptTextView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.quiz_history_activity);
+
+        ArrayList<History> histories = this.getIntent().getParcelableArrayListExtra("HISTORY");
+        averageTextView = (TextView) findViewById(R.id.quiz_history_average);
+        attemptTextView = (TextView) findViewById(R.id.quiz_history_attempt);
+
+        if (histories.size() > 0) {
+            Double average = computeAverage(histories);
```

Comment:
why are we using the object Double instead of the primitive type? I think this might be the reason why you needed three parameters in the assert equals

Reply:
ohh okiee my bad

Reply:
if you think the average should be able to return null (if it's meaningful) then I would keep it as Double or else primitive type would make more sense